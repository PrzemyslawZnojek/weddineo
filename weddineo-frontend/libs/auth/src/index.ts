export * from './lib/auth.module';
export * from './lib/+state/auth.facade'
