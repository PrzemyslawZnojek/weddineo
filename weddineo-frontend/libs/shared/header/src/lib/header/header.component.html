<mat-toolbar color="primary">
  <mat-toolbar-row [fxFlex]="'row'" [fxLayoutAlign]="'space-between center'">
    <div>{{ 'title' | translate }}</div>
    <div [fxLayoutAlign]="'space-between center'">
      <span class="username">{{ (username$  | async)?.email }}</span>
      <span class="material-icons">
        account_circle
      </span>
    </div>
  </mat-toolbar-row>
</mat-toolbar>
