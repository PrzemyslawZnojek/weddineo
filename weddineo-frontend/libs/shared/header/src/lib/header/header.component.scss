.username {
  margin-right: 10px;
}

.menu-button {
  padding: 0;
  min-width: 20px;
}

.logo {
  cursor: pointer;
}
