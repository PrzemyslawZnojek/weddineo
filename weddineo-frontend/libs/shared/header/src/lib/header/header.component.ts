import { Component, OnInit } from '@angular/core';
import { AuthFacade } from '@weddineo-frontend/auth';

@Component({
  selector: 'weddineo-frontend-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss'],
})
export class HeaderComponent implements OnInit {
  username$ = this.authFacade.getFirebaseUser$;

  constructor(private authFacade: AuthFacade) {}

  ngOnInit(): void {}
}
