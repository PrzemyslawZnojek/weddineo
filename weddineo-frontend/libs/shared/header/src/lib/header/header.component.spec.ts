import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FlexLayoutModule } from '@angular/flex-layout';
import { MockModule, MockPipe } from 'ng-mocks';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatIconModule } from '@angular/material/icon';

import { HeaderComponent } from './header.component';
import { TranslatePipe } from '@ngx-translate/core';
import { of } from 'rxjs';
import { AuthFacade } from '@weddineo-frontend/auth';

describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        FlexLayoutModule,
        MockModule(MatToolbarModule),
        MockModule(MatIconModule),
      ],
      declarations: [HeaderComponent, MockPipe(TranslatePipe)],
      providers: [{
        provide: AuthFacade,
        useClass: class {
          username$ = of({
            email: 'test@mail.com'
          })
        }
      }]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render username and icon', () => {
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    expect(compiled.getElementsByClassName('username').length).toEqual(1);
    expect(compiled.getElementsByClassName('material-icons').length).toEqual(1);
  });
});
