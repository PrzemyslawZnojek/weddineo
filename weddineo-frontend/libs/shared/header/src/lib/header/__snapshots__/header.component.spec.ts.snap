// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match snapshot 1`] = `
<weddineo-frontend-header
  authFacade={[Function useClass]}
  username$="undefined"
>
  <mat-toolbar
    color="primary"
    style="box-sizing: border-box; display: flex; flex-direction: row;"
  >
    <mat-toolbar-row
      style="align-content: center; align-items: center; box-sizing: border-box; display: flex; flex-direction: row; justify-content: space-between; flex: 1 1 100%; max-width: row;"
    >
      <div>
        
      </div>
      <div
        style="align-content: center; align-items: center; box-sizing: border-box; display: flex; flex-direction: row; justify-content: space-between;"
      >
        <span
          class="username"
        >
          
        </span>
        <span
          class="material-icons"
        >
           account_circle 
        </span>
      </div>
    </mat-toolbar-row>
  </mat-toolbar>
</weddineo-frontend-header>
`;
